comp/forwarder/defaultforwarder/worker.go

```diff
@@ -61,7 +65,13 @@ func NewWorker(
 	blocked *blockedEndpoints,
 	pointSuccessfullySent PointSuccessfullySent,
 	isLocal bool,
+	maxConcurrentRequests int64,
 ) *Worker {
+	var maxConcurrentRequestsSem *semaphore.Weighted
+	if maxConcurrentRequests > 0 {
+		maxConcurrentRequestsSem = semaphore.NewWeighted(maxConcurrentRequests)
```

Comment:
Is there a scenario when it makes sense to disable this limit? When disabled, this will create a new goroutine for every transaction (inside http2 lib) that would block for unspecified time once the number of in flight requests goes above connection limit and break our high/low prioritization of requests.

Reply:
There is a cost involved with acquiring the semaphore that maybe users might want to avoid. Granted, avoiding the semaphore exposes you to other costs, but that's the only real scenario I can think of.. Other than that you can just set this value to a really large number if you really want no limit.

Reply:
Reflecting on it, I have changed it now, so rather than disabling the limit if you set this to a value less than 1 it will warn and set the value to 1.